azure-spring-cloud-maven-plugin/src/main/java/com/microsoft/azure/maven/springcloud/ConfigMojo.java

```diff
@@ -310,24 +311,21 @@ private void configureAppName() throws IOException, InvalidConfigurationExceptio
     }
 
     private void selectAppCluster() throws IOException, InvalidConfigurationException {
-        final List<ServiceResourceInner> clusters = getSpringServiceClient().getAvailableClusters();
-
-        this.wrapper.putCommonVariable("clusters", clusters);
+        final AzureSpringCloud az = AzureSpringCloud.az(this.getClient());
         if (StringUtils.isNotBlank(clusterName)) {
-            final ServiceResourceInner clusterByName = clusters.stream().filter(t -> StringUtils.equals(this.clusterName, t.name())).findFirst()
-                    .orElse(null);
-            if (clusterByName != null) {
-
-                this.appSettings.setClusterName(clusterByName.name());
+            final SpringCloudCluster cluster = az.cluster(this.clusterName);
+            if (cluster.exists()) {
+                this.appSettings.setClusterName(cluster.entity().getName());
                 return;
             }
             getLog().warn(String.format("Cannot find Azure Spring Cloud Service with name: %s.", TextUtils.yellow(this.clusterName)));
         }
-
-        final ServiceResourceInner targetAppCluster = this.wrapper.handleSelectOne("select-ASC", clusters, null, ServiceResourceInner::name);
+        final List<SpringCloudCluster> clusters = az.clusters();
+        this.wrapper.putCommonVariable("clusters", clusters);
+        final SpringCloudCluster targetAppCluster = this.wrapper.handleSelectOne("select-ASC", clusters, null, c -> c.entity().getName());
         if (targetAppCluster != null) {
-            this.appSettings.setClusterName(targetAppCluster.name());
-            getLog().info(String.format("Using service: %s", TextUtils.blue(targetAppCluster.name())));
+            this.appSettings.setClusterName(targetAppCluster.entity().getName());
```

Comment:
can we create an method named getName() in SpringCloudCluster  since it is the very common property